src/clr-angular/layout/vertical-nav/vertical-nav-link.ts

```diff
@@ -1,28 +1,35 @@
 /*
- * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import { Component, HostListener, Optional } from '@angular/core';
+import { Component, HostListener, Optional, ElementRef } from '@angular/core';
 import { VerticalNavGroupService } from './providers/vertical-nav-group.service';
 
 @Component({
   selector: '[clrVerticalNavLink]',
   template: `
         <ng-content select="[clrVerticalNavIcon]"></ng-content>
         <span class="nav-text">
-            <ng-content></ng-content>    
+            <ng-content></ng-content>
         </span>
     `,
-  host: { class: 'nav-link' },
+  host: {
+    class: 'nav-link',
+    '[attr.aria-current]': 'setAriaCurrent',
+  },
 })
 export class ClrVerticalNavLink {
-  constructor(@Optional() private _navGroupService: VerticalNavGroupService) {}
+  constructor(@Optional() private _navGroupService: VerticalNavGroupService, private el: ElementRef) {}
 
   @HostListener('click')
   public expandParentNavGroup(): void {
     if (this._navGroupService) {
       this._navGroupService.expand();
     }
   }
+
+  get setAriaCurrent() {
+    return this.el.nativeElement.classList.contains('active') ? 'page' : null;
```

Comment:
Two issues, this still runs with every change detection cycle and it isn't guarded with a platform check. If we can't find a better option, then we should just change this to document that applications should add the `aria-current` value directly.

Reply:
It won't actually run every change detection cycle. We are accessing DOM api so it's idle, and it won't respond even if the `active` class is toggled. 

@bdryanovski another small issue is in the naming of this getter property. You have `getter` but the name is `setAriaCurrent` which usually suggests a method from API perspective :D. We could rename this property to `isActive`.

Reply:
@Shijir you are right, the name and the implementation changed over the time, so the name need to be changed as well. I will changed.

@gnomeontherun you are right for the platform check, will add it. When it comes to implementing it from outside it will be more heavy on the performance if you have to do DOM selection or going over it with reference. Class list is updated with the related component so this is done no matter what. 

It could be done in the same way with @HostBinding but decide to use the host syntax just to be a lot more clear and compact. 

One issue with that is that .nav-link and .active are added dynamically and angular is having hard time tracking them, without making DOM queries, so I find this to be the most effective way to do it. 

We could go with documentation only, but I don’t think that this is adding a lot of performance penalty.

Reply:
It is running every change detection cycle. Nothing about the binding is linked to the `active` class so Angular will check each time. When you click a button it fires or add an interval and see that it fires the binding on every interval. 

Host bindings are always checked, so if their values have to be looked up, that lookup happens with each change detection. The downstream change (setting of aria-current) only changes if the lookup returns a different value. Regardless if a change is found, the DOM is still hit for _every_ vertical nav link and _every_ change detection cycle. 

I hadn't thought through what a better option would be, but this is clearly a blocker as is. Based on this, we should abandon handling it internally and document how applications should do this. The other concern is that vertical-nav _may_ be used in ways that aren't pages so the aria-current implementation might not make sense if the application is using it like a tabbed interface. They shouldn't do this probably, but it is a concern. Either way, we've tried several things and they all suffer from performance drains that we can't accept, so let's go to documentation and close this out.

Reply:
If this is the case, will revert the changes for the ClrVericalNavLink and will just update the documentation.